connect-inject/handler.go

```diff
@@ -393,6 +478,76 @@ func (h *Handler) defaultAnnotations(pod *corev1.Pod, patches *[]jsonpatch.JsonP
 	return nil
 }
 
+func (h *Handler) consulNamespace(ns string) string {
+	// Return an empty string if namespaces aren't enabled. This
+	// functionality is used for init and sidecar container creation.
+	if !h.EnableNamespaces {
+		return ""
+	}
+
+	// Mirroring takes precedence
+	if h.EnableNSMirroring {
+		return fmt.Sprintf("%s%s", h.MirroringPrefix, ns)
+	} else {
+		return h.ConsulNSName
+	}
+}
+
+func (h *Handler) checkAndCreateNamespace(ns string) error {
+	// Check if the Consul namespace exists
+	namespaceInfo, _, err := h.ConsulClient.Namespaces().Read(ns, nil)
+	if err != nil {
+		return err
+	}
+
+	// If not, create it
+	if namespaceInfo == nil {
+		consulNamespace := api.Namespace{
+			Name:        ns,
+			Description: "Auto-generated by a Connect Injector",
```

Comment:
```suggestion
			Description: "Auto-generated by Connect Injector",
```

Reply:
I phrased it this way because namespaces cut across datacenters, so if someone is running multiple k8s clusters with various injectors, only one of them will have created it. Maybe that's pedantic and potentially confusing, but technically more accurate.